### DISCUSSION TITLE
BurntToast Button to reboot the computer

### DISCUSSION THREAD

**Discussion by tomsoe** [Windos/BurntToast#140] [Q&A]
Is it possible to execute a powershell command when clicking a button with BurntToast now? I saw the 'Toast Notifications with PowerShell: From Noob to Master Baker' on Youtube recently where the example was "Write-Warning". Unfortunately I'm not a Master Baker..yet.

I want to create a script that lets the users reboot their computer by cliking "Restart" when the Toast shows up, if they're busy they can choose "Dismiss".

**Comment by sgexcit** [comment]
One of the contributors to this project [KelvinTegelaar](https://github.com/KelvinTegelaar) has their own blog referencing this with the exact scenario you are looking for that I'm sure could be helpful: https://www.cyberdrain.com/monitoring-with-powershell-notifying-users-of-windows-updates/

He used a registry edit in the script to make a protocol handler for the reboot command.

edit: edited the url

**Reply by Windos** [reply]
This is very much the way to go unless you're able to run PowerShell 7.1 everywhere (which I'm not ☹️)

**Comment by tomsoe** [comment]
@sgexcit Thanks for providing a solution. :) 
..However I'd like to try this method first. It is based on some of the latest presentations of Windos' BurntToast on Youtube that I could find, and I liked this approach better than creating a protocol handler by editing registry: 

```
$Activated = {
    if($Event.SourceArgs[1].Arguments -eq 'dismiss'){
        Write-Warning 'User clicked the restart button, initiating reboot'
        Start-Process 'C:\windows\system32\notepad.exe'
        #Restart-Computer
    } elseif ($Event.SourceArgs[1].Arguments -eq 'snooze') {
            Write-Warning 'User clicked the snooze, toast will pop up in default time'
        } else {
            Write-Warning 'User clicked on the toast and not the button'   
        }      
}

$Dismissed = {
    if($Event.SourceArgs[1].Reason -eq 'TimedOut'){
        Write-Warning 'The Toast timed out..'
        #$Global:ToastEvent = $Event
    } 
    if($Event.SourceArgs[1].Reason -eq 'UserCanceled') {
        Write-Warning 'User canceled the toast!'
       #$Global:ToastEvent = $Event
    }
}

$ActivateButton     = New-BTButton -Dismiss -Content "Restart" 
$DismissButton      = New-BTButton -Snooze -Content "Senere" 

$image = "----" ##Logo

$ToastBody = @{
    Text    = "$HeaderText", "$ToastText1", "$ToastText2"
    Applogo = $image
    Sound   = 'Default'

}

 New-BurntToastNotification @ToastBody -ActivatedAction $Activated -DismissedAction $Dismissed -Button $ActivateButton, $DismissButton 
```

**Reply by Windos** [reply]
Just keep in mind, that in order for this to work it needs to be in PowerShell 7.1 or above. Unfortunately, before that the protocol handler is the way to go.

**Reply by tomsoe** [reply]
@Windos Ah, that's unfortunate.. Well then I guess my initial script will be set on-hold for now. 
My plan is to implement the reboot script to a proactive remediation script in Azure/Intune that forces users to reboot their computer if the uptime is more than x days. :)

I'm not sure which version of PowerShell is in Azure, but I guess the script will be lauched with the current users version of PS, and I know for a fact that's not 7.1..

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
